File: Easy/Reverse Bits.cpp
//{ Driver Code Starts
#include <bits/stdc++.h>
using namespace std;

// } Driver Code Ends
class Solution
{
  public:
    long long reversedBits(long long X)
    {
        // code here
        string binario = bitset<32>(X).to_string();
        int n = binario.length();
        
        long long ans = 0;
        int p = 0;
        for (int i=0; i<n; i++)
        {
            // cout << binario[i] << " " << binario[i] - '0' << endl;
            ans += (binario[i] - '0') * pow (2, p);
            p++;
        }
        
        return ans;
    }
};

//{ Driver Code Starts.
int main() {
    int t;
    cin >> t;
    while (t--) {
        long long X;
        
        cin>>X;

        Solution ob;
        cout << ob.reversedBits(X) << endl;
    }
    return 0;
}
// } Driver Code Ends

File: Easy/Reverse Bits.py
#User function Template for python3
import math

class Solution:
    def reverseBits(self, n):
        #code here
        bin = []
        
        while n > 0:
            bin += [n % 2]
            n //= 2
        
        ans = 0
        l = len(bin)
        
        j = 0
        for i in range(l-1, -1, -1):
            digit = bin[i] %2
            if (digit == 1):
                ans += math.pow(2, j)
            j += 1
            #print(digit)
        
        return int(ans)

#{ 
 # Driver Code Starts
#Initial Template for Python 3

import math

def main():
    
    T=int(input())
    
    while(T>0):
        
        
        n=int(input())
        
        ob=Solution()
        print(ob.reverseBits(n))
     
        T-=1

if __name__=="__main__":
    main()
# } Driver Code Ends